### PR TITLE
ONEUP-3283: Add Highlight Stage channel

### PR DESCRIPTION
**Pull request by Sebastian-Fitzner**

### CODE REVIEW

**Review by Normal — APPROVED** [review]

File: raw/src/test/scala/de/welt/contentapi/raw/models/RawWritesTest.scala [hunk 1]
```diff
@@ -373,7 +373,73 @@ class RawWritesTest extends PlaySpec {
...
+        index = 3,
+        hidden = true,
+        trackingName = Some("track"),
+        link = Some(RawSectionReference(label = Some("label"), path = Some("/path"))),
+        highlightSectionMapping = "high-section",
+        highlightStageMapping = "high-stage",
+        label = "Name",
+        layout = HighlightStageLayout.HERO_CAROUSEL,
+        logo = Some("/logo/a"),
+        outroTeaser = Some(OutroTeaser(link = "o-link", label = "o-label", topic = "o-topic")),
```
Comment (aramin) [inline raw/src/test/scala/de/welt/contentapi/raw/models/RawWritesTest.scala:412]: should it really be used with class names in the end?

File: raw/src/main/scala/de/welt/contentapi/raw/models/RawChannel.scala [hunk 2]
```diff
@@ -506,6 +509,41 @@ case class RawChannelStageCurated(override val index: Int,
...
+                                    override val hidden: Boolean = RawChannelStage.HiddenDefault,
+                                    override val trackingName: Option[String] = None,
+                                    override val link: Option[RawSectionReference] = None,
+                                    highlightSectionMapping: String,
+                                    highlightStageMapping: String,
+                                    label: String,
+                                    layout: HighlightStageLayout = HighlightStageLayout.FULL_CAROUSEL,
+                                    logo: Option[String] = None,
+                                    outroTeaser: Option[OutroTeaser] = None,
+                                    references: Seq[RawSectionReference] = Seq.empty,
```
Comment (aramin) [inline raw/src/main/scala/de/welt/contentapi/raw/models/RawChannel.scala:543]: as i already mentioned in the cmcf pr, we need to support multiple types of references (see https://github.com/spring-media/rbbt-ConfigMcConfigFace/pull/318#discussion_r646694793)

**Review by aramin — APPROVED** [review]

File: raw/src/main/scala/de/welt/contentapi/raw/models/RawChannel.scala (hunk 2 above)
Comment (Sebastian-Fitzner) [reply raw/src/main/scala/de/welt/contentapi/raw/models/RawChannel.scala:543]: @aramin we talked about this already and are adding another option in SectionReference to support path or tracking report name... I am currently implementing it.

File: raw/src/test/scala/de/welt/contentapi/raw/models/RawReadsTest.scala [hunk 3]
```diff
@@ -214,7 +212,80 @@ class RawReadsTest extends PlaySpec {
...
+            "hidden": true,
+            "trackingName": "track-it",
+            "link": { "label": "Sport Link" , "path": "/sport" },
+            "highlightSectionMapping": "frontpage",
+            "highlightStageMapping": "sondergruppe-1",
+            "label": "My custom label",
+            "layout": "HERO_CAROUSEL",
+            "logo": "/icon/winner",
+            "outroTeaser": { "link": "/sport/teaser", "label": "Sport Teaser", "topic": "all videos" },
+            "teaserOutroLink": "/sport/fussball",
```
Comment (Sebastian-Fitzner) [inline raw/src/test/scala/de/welt/contentapi/raw/models/RawReadsTest.scala:265]: do we need these fields when `outroTeaser` is already available?

File: raw/src/test/scala/de/welt/contentapi/raw/models/RawWritesTest.scala (hunk 1 above)
Comment (catalin-cretu) [reply raw/src/test/scala/de/welt/contentapi/raw/models/RawWritesTest.scala:412]: You mean the test values? These are just dummy values.

Comment (aramin) [reply raw/src/test/scala/de/welt/contentapi/raw/models/RawWritesTest.scala:412]: more realistic values could be helpful to understand what this field is used for

**Pull request merged**